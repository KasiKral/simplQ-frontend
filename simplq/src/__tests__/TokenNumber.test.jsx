import React from 'react';
import { render, fireEvent, cleanup, getByTitle } from '@testing-library/react';
import '@testing-library/jest-dom';
import Token from '../components/pages/Status/TokenNumber';

afterEach(cleanup);

it('renders correctly', () => {
    const { asFragment } = render(<Token />);
    expect(asFragment()).toMatchSnapshot();
});

it('generate token on page', () => {
    const { getByText, getByRole } = render(<Token tokenNumber="1234" />);
    expect(getByText("1234")).toBeTruthy();
});

it("component's elements are setup correctly", () => {
    const { container } = render(<Token />);;
    expect(container.firstChild).toHaveClass('token-number');
});