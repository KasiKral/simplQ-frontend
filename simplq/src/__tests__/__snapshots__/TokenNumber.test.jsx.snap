// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="token-number"
  >
    <p>
      Token No: 
    </p>
    <p
      class="count"
    />
  </div>
</DocumentFragment>
`;
